app/models/vm.rb

```diff
@@ -1,6 +1,7 @@
 class Vm < VmOrTemplate
   default_scope { where(:template => false) }
-
+  has_one :container_deployment, :through => :container_deployment_node
```

Comment:
thats right, but its not enforce with the DB.
e.g.
```ruby
ContainerDeploymentNode.create(vm_id: 1, container_deployment:1)
ContainerDeploymentNode.create(vm_id: 1, container_deployment:2)
Vm.find(1). container_deployment_node == ?
```

You could avoid such uncertainty by adding an uniq index(vm_id) to ContainerDeploymentNode and/or adding a validation to ContainerDeploymentNode